### PR TITLE
feat(express-serve-static-core): Add Layer typings to express Router & Route stack prop

### PR DESCRIPTION
**Pull request by douglasg14b**
## Relevant source for Layer:
- Layer source: https://github.com/expressjs/express/blob/master/lib/router/layer.js#L142
- Router Usage for `route` property: https://github.com/expressjs/express/blob/
4b9cd2fd0e13519a16bc36e8b4212e7924698b2e/lib/router/index.js#L227
   - `route` assignment: https://github.com/expressjs/express/blob/4b9cd2fd0e13519a16bc36e8b4212e7924698b2e/lib/router/index.js#L511
   - `route` assignment to `undefined`: https://github.com/expressjs/express/blob/4b9cd2fd0e13519a16bc36e8b4212e7924698b2e/lib/router/index.js#L481
- Route `stack` prop: https://github.com/expressjs/express/blob/4b9cd2fd0e13519a16bc36e8b4212e7924698b2e/lib/router/route.js#L200
- Router `stack` prop: https://github.com/expressjs/express/blob/4b9cd2fd0e13519a16bc36e8b4212e7924698b2e/lib/router/index.js#L513

<hr>

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

File: types/express-serve-static-core/index.d.ts [hunk 1]
```diff
@@ -292,12 +292,22 @@ export interface IRouter extends RequestHandler {
...
+}
+
+export interface ILayer {
+    route: IRoute;
+    name: string || '<anonymous>';
+    params?: Record<string, any>;
+    keys: string[];
+    path?: string;
+    regexp: RegExp;
+    handle: (req: Request, res: Response, next: NextFunction) => any
```
Comment (douglasg14b) [inline types/express-serve-static-core/index.d.ts:305]:
```suggestion
    handle: (req: Request, res: Response, next: NextFunction) => any;
```

File: types/express-serve-static-core/index.d.ts [hunk 2]
```diff
@@ -292,12 +292,22 @@ export interface IRouter extends RequestHandler {
     /**
      * Stack of configured routes
      */
-    stack: any[];
+    stack: ILayer[];
+}
+
+export interface ILayer {
+    route: IRoute;
```
Comment (douglasg14b) [inline types/express-serve-static-core/index.d.ts:299]: `route` can be `undefined` 

https://github.com/expressjs/express/blob/4b9cd2fd0e13519a16bc36e8b4212e7924698b2e/lib/router/index.js#L481

```suggestion
    route?: IRoute;
```

**Comment by douglasg14b** [comment]
This PR appears to be more detailed, but was closed: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/63329

File: types/express-serve-static-core/index.d.ts [hunk 3]
```diff
@@ -292,12 +292,22 @@ export interface IRouter extends RequestHandler {
     /**
      * Stack of configured routes
      */
-    stack: any[];
+    stack: ILayer[];
+}
+
+export interface ILayer {
+    route?: IRoute;
+    name: string || '<anonymous>';
```
Comment (douglasg14b) [inline types/express-serve-static-core/index.d.ts:300]:
```suggestion
    name: string | '<anonymous>';
```

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/express-serve-static-core/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 4ceb95c9f955816da9e970edbed563516a2465b3
</p>

**Review by sheetalkamat — APPROVED** [review]

**Comment by douglasg14b** [comment]
Ready to merge

**Pull request merged**